packages/cliqz-home-react/src/Pagination.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component, PropTypes } from 'react';
+
+const propTypes = {
+  items: PropTypes.array.isRequired,
+  onChangePage: PropTypes.func.isRequired,
+  initialPage: PropTypes.number
+}
+
+const defaultProps = {
+  initialPage: 1
+}
+
+class Pagination extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      pager: {}
+    }
+    this.timer = null;
+  }
+
+  componentWillMount() {
+    if (this.props.items && this.props.items.length) {
+      this.setPage(this.props.initialPage);
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.items !== prevProps.items) {
+      this.setPage(this.props.initialPage);
+    }
+  }
+
+  autoRotate() {
+    clearInterval(this.timer);
+    this.timer = setInterval( () => {
+      const pager = this.state.pager;
+      const currentPage = this.state.pager.currentPage;
+      const totalPages = this.state.pager.totalPages;
+      if (currentPage === totalPages) {
+        pager.currentPage = 1
+      } else {
+        pager.currentPage = currentPage + 1;
+      }
+      this.setState({
+        pager: pager
+      });
+      this.setPage(this.state.pager.currentPage);
+    }, 5000);
+  }
+
+  animate(setNextPage) {
+    clearInterval(this.timer);
+    this._fadeOut().then(() => {
+      setNextPage();
+      this._fadeIn();
+    });
+  }
+
+  setPage(page) {
+    this.animate( () => {
+      this._setNextPage(page);
+    });
+    this.autoRotate();
+  }
+
+  getPager(totalItems, currentPage, pageSize) {
+    currentPage = currentPage || 1;
+    pageSize = pageSize || 3;
+    var totalPages = Math.ceil(totalItems / pageSize);
+    var startIndex = (currentPage - 1) * pageSize;
+    var endIndex = Math.min(startIndex + pageSize - 1, totalItems - 1);
+
+    var pages = [];
+    for(var i = 1; i<= totalPages; i++) {
```

Comment:
would prefer this kind of implementation `[...Array(totalPages).keys()].map(i => i+1)`